arch/arm/core/aarch64/cpu_idle_qemu.c

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+ /**
+  * @file
+  *
+  * @brief Workaround aarch64 QEMU not responding to host OS signals
+  * during 'wfi'.
+  * See https://github.com/zephyrproject-rtos/sdk-ng/issues/255
+  */
+
+#include <arch/cpu.h>
+
+void arch_cpu_idle(void)
+{
+	/* Do nothing but unconditionally unlock interrupts and return to the
+	 * caller. We write 0 to the DAIF register, which un-masks all
+	 * interrupts.
+	 */
+	arch_irq_unlock(0);
```

Comment:
any reason why we want to unmask also FIQs & co? In the original code we only deal with IRQs (`DAIFSET_IRQ`).

Reply:
in this context we don't know what the original IRQ state is, we have to pick a hard-coded value to restore to.

do you mask FIQs when the idle thread is running?

Reply:
> in this context we don't know what the original IRQ state is, we have to pick a hard-coded value to restore to.

Can't we just assume that we had only the IRQs masked and unlock with `arch_irq_unlock(DAIF_IRQ)`?

> do you mask FIQs when the idle thread is running?

FIQs are currently always masked.

Reply:
> Can't we just assume that we had only the IRQs masked and unlock with `arch_irq_unlock(DAIF_IRQ)`?

The key parameter is expected to be the DAIF register state *before* the matching irq_lock call was made.
If I add as written above, that will expand to

```
msr daif, (1 << 7)
```

Which will result in all interrupts being unmasked...except the IRQs, essentially doing the opposite of what we want.

Let's keep in mind here this is a workaround for the emulator and will never end up on HW

Reply:
If you want me to change the constant value of `0` here, need to provide what you expect the default DAIF state to be when the idle thread is not sleeping.

Reply:
Yes, I had a brain fart on that and you are right that QEMU is unlikely to raise any FIQs really, still I wonder if we can just have instead something like `msr     daifclr, #(DAIFSET_IRQ)`

Reply:
let me try that really quick

Reply:
yup that seems to work, updated. thanks for reviewing